### PR TITLE
fix: fix some permission bugs

### PR DESCRIPTION
**Pull request by ybzky**
close #xxx

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

File: packages/sheets-ui/src/controllers/permission/sheet-permission-init.controller.ts [hunk 1]
```diff
@@ -280,12 +283,6 @@ export class SheetPermissionInitController extends Disposable {
         this._univerInstanceService.getAllUnitsForType<Workbook>(UniverInstanceType.UNIVER_SHEET).forEach((workbook) => {
             initSheetPermissionFunc(workbook);
         });
-
-        this.disposeWithMe(
-            this._univerInstanceService.getTypeOfUnitAdded$<Workbook>(UniverInstanceType.UNIVER_SHEET).subscribe((workbook) => {
```
Comment (Gggpound) [inline packages/sheets-ui/src/controllers/permission/sheet-permission-init.controller.ts:285]: 以后新增的咋办？

**Review by Gggpound — APPROVED** [review]

File: packages/sheets-ui/src/controllers/permission/sheet-permission-init.controller.ts (hunk 1 above)
Comment (ybzky) [reply packages/sheets-ui/src/controllers/permission/sheet-permission-init.controller.ts:285]: 在新增表格的地方已经初始化

File: packages/sheets/src/services/permission/range-permission/range-protection.ref-range.ts [hunk 2]
```diff
@@ -49,13 +47,16 @@ import {
...
+import type { IAddRangeProtectionMutationParams } from '../../../commands/mutations/add-range-protection.mutation';
+import { AddRangeProtectionMutation } from '../../../commands/mutations/add-range-protection.mutation';
 
 const mutationIdByRowCol = [InsertColMutation.id, InsertRowMutation.id, RemoveColMutation.id, RemoveRowMutation.id];
 const mutationIdArrByMove = [MoveRowsMutation.id, MoveColsMutation.id];
 
 type IMoveRowsOrColsMutationParams = IMoveRowsMutationParams;
 
-@OnLifecycle(LifecycleStages.Starting, RangeProtectionRefRangeService)
+@OnLifecycle(LifecycleStages.Steady, RangeProtectionRefRangeService)
```
Comment (Gggpound) [inline packages/sheets/src/services/permission/range-permission/range-protection.ref-range.ts:59]: 这里滞后了，初始化的时候那些apply导致的变更，可能就无法监听了。

File: packages/sheets/src/services/permission/range-permission/range-protection.ref-range.ts [hunk 3]
```diff
@@ -161,8 +159,8 @@ export class RangeProtectionRefRangeService extends Disposable {
 
         const removeRange = params.range;
         if (permissionRangeLapRules.length) {
-            const redoMutations: { id: string; params: ISetRangeProtectionMutationParams }[] = [];
-            const undoMutations: { id: string; params: ISetRangeProtectionMutationParams }[] = [];
+            const redoMutations: IMutationInfo[] = [];
```
Comment (Gggpound) [inline packages/sheets/src/services/permission/range-permission/range-protection.ref-range.ts:162]: 参数加上类型，补全范型

File: packages/sheets/src/services/permission/worksheet-permission/worksheet-permission-point.model.ts [hunk 4]
```diff
@@ -94,4 +94,14 @@ export class WorksheetProtectionPointModel {
         }
         return subUnitMap;
     }
+
+    getTargetByPermissionId(unitId: string, permissionId: string) {
```
Comment (Gggpound) [inline packages/sheets/src/services/permission/worksheet-permission/worksheet-permission-point.model.ts:98]: 命名改一下，getUnit

File: packages/sheets/src/services/permission/worksheet-permission/worksheet-permission-rule.model.ts [hunk 5]
```diff
@@ -24,7 +24,7 @@ type IRuleChangeType = 'add' | 'set' | 'delete';
 export class WorksheetProtectionRuleModel {
     /**
      *
-     * Map<unitId, Map<subUnitId, Map<ruleId, IWorksheetProtectionRule>>>
```
Comment (Gggpound) [inline packages/sheets/src/services/permission/worksheet-permission/worksheet-permission-rule.model.ts:27]: 诶？这里的key改了？没看到其他改动

**Review by Gggpound — APPROVED** [review]

**Review by Gggpound — APPROVED** [review]

File: packages/sheets/src/services/permission/range-permission/range-protection.ref-range.ts [hunk 6]
```diff
@@ -108,13 +110,10 @@ export class RangeProtectionRefRangeService extends Disposable {
...
-                    const params = commandInfo.params as {
-                        permissionId: string;
-                        rule: IRangeProtectionRule;
-                    };
-                    const subUnitId = params.rule.subUnitId;
-                    const unitId = params.rule.unitId;
+                if (commandInfo.id === SetRangeProtectionMutation.id || commandInfo.id === AddRangeProtectionMutation.id) {
+                    const params = commandInfo.params as ISetRangeProtectionMutationParams | IAddRangeProtectionMutationParams;
+                    const subUnitId = params.subUnitId;
+                    const unitId = params.unitId;
```
Comment (Gggpound) [inline packages/sheets/src/services/permission/range-permission/range-protection.ref-range.ts:116]: 这里在无界模式的初始化 应该有问题，处理下 workbook的销毁创建，以及初始所有的workbook

**Pull request merged**